File: backend/src/main/java/com/example/hrm/service/SalaryService.java
package com.example.hrm.service;

import com.example.hrm.dto.salary.SalaryAssignmentRequest;
import com.example.hrm.dto.salary.SalaryStructureDto;

import java.util.List;

public interface SalaryService {
    SalaryStructureDto createStructure(SalaryStructureDto dto);
    List<SalaryStructureDto> getAllStructures();
    SalaryStructureDto getStructureById(Long id);
    SalaryStructureDto updateStructure(Long id, SalaryStructureDto dto);
    void deleteStructure(Long id);

    void assignSalaryToEmployee(SalaryAssignmentRequest request);
    SalaryStructureDto getActiveSalaryForEmployee(Long employeeId);
    List<SalaryStructureDto> getEmployeeSalaryHistory(Long employeeId);
}

File: frontend/src/services/salaryService.js
import api from './api';

export const salaryService = {
  getAllStructures: async () => {
    const res = await api.get('/salary-structures');
    return res.data.data;
  },
  createStructure: async (data) => {
    const res = await api.post('/salary-structures', data);
    return res.data.data;
  },
  updateStructure: async (id, data) => {
    const res = await api.put(`/salary-structures/${id}`, data);
    return res.data.data;
  },
  assignSalary: async (data) => {
    const res = await api.post('/salary-structures/assign', data);
    return res.data;
  },
  getActiveSalaryForEmployee: async (employeeId) => {
    const res = await api.get(`/salary-structures/employee/${employeeId}`);
    return res.data.data;
  },
  getSalaryHistoryForEmployee: async (employeeId) => {
    const res = await api.get(`/salary-structures/employee/${employeeId}/history`);
    return res.data.data;
  },
};
